xpath.d.ts

```diff
@@ -1,11 +1,35 @@
 /// <reference lib="dom" />
 
-type SelectedValue = Node | Attr | string | number | boolean;
-interface XPathSelect {
-    (expression: string, node?: Node): Array<SelectedValue>;
-    (expression: string, node: Node, single: true): SelectedValue | undefined;
+export type SelectedValue = Node | string | number | boolean | null;
+
+export interface XPathSelect {
+    (expression: string, node: Node): Array<Node>;
+    (expressed: string, node: Node, single: false): Array<Node>;
+    (expression: string, node: Node, single: true): SelectedValue;
 }
-export var select: XPathSelect;
-export function select1(expression: string, node?: Node): SelectedValue | undefined;
-export function evaluate(expression: string, contextNode: Node, resolver: XPathNSResolver | null, type: number, result: XPathResult | null): XPathResult;
-export function useNamespaces(namespaceMap: { [name: string]: string }): XPathSelect;
+
+export type SelectWithResolverReturnType = Array<Node> | SelectedValue;
+export type SelectWithResolverSingleReturnType = SelectedValue;
```

Comment:
Is there a reason to use this alias instead of using `SelectedValue` directly? Its use on line 15 doesn't really make sense because that `select()` function doesn't involve a resolver.

Reply:
I've renamed it. I'm not using `SelectedValue` directly because of uniformity. If you'd prefer it to just be `SelectedValue`, I'll gladly make the change.

Reply:
Ok, I think these names are good.